src/braket/pennylane_plugin/ahs_translation.py

```diff
@@ -195,3 +200,134 @@ def _get_sample_times(time_interval: ArrayLike):
 
     # we return time in seconds
     return times / 1e9
+
+
+def _create_valid_local_detunings(local_pulses, dev_wires):
+    """Return ordered list of local detunings for all wires in device.
+
+    This function uses the local detunings of the pulses of the ``ParametrizedEvolution`` being
+    executed to create a list of local detunings with the same length and order as the device
+    wires. For wires that aren't locally detuned, the list is padded with zeros if the detunings
+    are of type ``float``, or functions that return zero if the detunings are ``callable``.
+
+    Args:
+        local_pulses (List[HardwarePulse]): Partially evaluated list of local pulses
+        dev_wires (~.Wires): Device wires
+
+    Returns:
+        List[Union[callable, float]]: List of detunings covering all device wires.
+    """
+    if len(local_pulses) == 0:
+        return None
+
+    callable_detunings = callable(local_pulses[0].frequency)
+    device_detunings = (
+        [lambda t: 0] * len(dev_wires) if callable_detunings else [0] * len(dev_wires)
+    )
+
+    for pulse in local_pulses:
+        for wire in pulse.wires:
+            device_detunings[dev_wires.index(wire)] = pulse.frequency
+
+    return device_detunings
+
+
+def _extract_pattern_from_detunings(detunings, time_points):
+    """Use the detunings to find the pattern for the ``ShiftingField``.
+
+    This function creates a time series for the local detunings and uses the values
+    of the detunings at each time step to calculate and validate the pattern for the
+    ``ShiftingField`` term of the driving Hamiltonian.
+
+    Args:
+        detunings (List[Union[float, callable]]): detunings to extract pattern from
+        time_points (array[Number, Number]]): Array of sampled time steps
+
+    Returns:
+        Union[float, callable]: Maximum detuning to be used as the magnitude for ``ShiftingField``
+        Pattern: object containing magnitude of detunings for individual atoms in the device
+
+    Raises:
+        ValueError: if the shape of all local detunings don't match or the detunings have negative
+        values
+    """
+    # If a single item is not callable, no others should be callable. This validation happens in
+    # ``_validate_pulses``.
+    callable_detunings = callable(detunings[0])
+    negative_detunings_error = (
+        "Found negative value in local detunings. Make sure that all local detunings "
+        + "take only positive values within the specified time interval."
+    )
+
+    if callable_detunings:
+        evaluated_detunings = np.array(
+            [[float(detuning(t * 1e6)) for t in time_points] for detuning in detunings]
```

Comment:
From the test failure, it looks like this is calling a staticmethod as a standalone function, which wasn't introduced to Python [until 3.10](https://docs.python.org/3/whatsnew/3.10.html#other-language-changes):
> Static methods ([@staticmethod](https://docs.python.org/3/library/functions.html#staticmethod)) and class methods ([@classmethod](https://docs.python.org/3/library/functions.html#classmethod)) now inherit the method attributes (__module__, __name__, __qualname__, __doc__, __annotations__) and have a new __wrapped__ attribute. Moreover, static methods are now callable as regular functions. (Contributed by Victor Stinner in [bpo-43682](https://bugs.python.org/issue?@action=redirect&bpo=43682).)